server/sonar-server/src/main/java/org/sonar/server/branch/ws/ListAction.java

```diff
@@ -96,30 +97,35 @@ public void handle(Request request, Response response) throws Exception {
       userSession.checkComponentPermission(UserRole.USER, project);
       checkArgument(project.isEnabled() && PROJECT.equals(project.qualifier()), "Invalid project key");
 
-      Collection<BranchDto> branches = dbClient.branchDao().selectByComponent(dbSession, project);
+      Collection<BranchDto> branches = dbClient.branchDao().selectByComponent(dbSession, project).stream()
+        // TODO replace by a check on the key_type, when available
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fbranch%2Ffirst-class-pull-request&issues=AWGZgq7Qqu6zmfejgj10&open=AWGZgq7Qqu6zmfejgj10)

[ISSUE_KEY]: AWGZgq7Qqu6zmfejgj10